processing/src/main/java/io/druid/segment/column/ColumnCapabilitiesImpl.java

```diff
@@ -26,12 +26,12 @@
  */
 public class ColumnCapabilitiesImpl implements ColumnCapabilities
 {
-  private ValueType type = null;
-  private boolean dictionaryEncoded = false;
-  private boolean runLengthEncoded = false;
-  private boolean hasInvertedIndexes = false;
-  private boolean hasSpatialIndexes = false;
-  private boolean hasMultipleValues = false;
+  private volatile ValueType type = null;
```

Comment:
would be nice to write a header for this file about thread safety and how it can be used